### PR TITLE
refactor: Change PhysicsSolver to PhysicsPackage

### PR DESCRIPTION
**Pull request by rrsettgast**
Changes:
- physicsSolvers -> physicsPackages
- SolverBase -> PhysicsPackageBase
- PhysicsSolverManager -> PhysicsPackageManager
- ContactSolverBase -> ContactPackageBase
- FlowSolverBase -> FlowPackageBase

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -28,16 +28,16 @@
...
+/src/coreComponents/physicsPackages/fluidFlow          @CusiniM @paveltomin @klevzoff @francoishamon
+/src/coreComponents/physicsPackages/inducedSeismicity  @CusiniM @rrsettgast
+/src/coreComponents/physicsPackages/multiphysics       @CusiniM @paveltomin @castelletto1 @rrsettgast
+/src/coreComponents/physicsPackages/python             @cssherman @corbett5
+/src/coreComponents/physicsPackages/simplePDE          @castelletto1 @rrsettgast @CusiniM 
+/src/coreComponents/physicsPackages/solidMechanics     @rrsettgast @castelletto1 @CusiniM 
+/src/coreComponents/physicsPackages/surfaceGeneration  @rrsettgast @CusiniM @jhuang2601
+/src/coreComponents/physicsPackages/wavePropagation    @sframba @acitrain
 /src/coreComponents/schema
 /src/coreComponents/unitTests                         @rrsettgast @corbett5 @wrtobin
```
Comment (rrsettgast) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/unitTests                         @rrsettgast @CusiniM 
```

**Review by acitrain — CHANGES_REQUESTED** [review]
Hi Randy,

I tried to compile on Pangea 3 (only GEOS, units tests are currently broken due to a previous merge see PR #3244 ), and I found a remaining SolverBase inside 

src/coreComponents/physicsPackages/python/PySolver.cpp : geos::SolverBase * group; it is on line 43

**Review by CusiniM — APPROVED** [review]
It's pretty hard to review but it looks good. I think it would be nice to also collect all kernel files for the flow solvers in a subdir of fluidFlow like we did for the mechanics ones.

**Review by acitrain — APPROVED** [review]
Seems good ! 
Thanks Randy!

**Comment by castelletto1** [comment]
Now that we got rid of the concept of `physics solver`, what is the meaning of function `solverStep` in `PhysicsPackageBase.hpp`? Maybe we should rename it.

**Comment by rrsettgast** [comment]
> Now that we got rid of the concept of `physics solver`, what is the meaning of function `solverStep` in `PhysicsPackageBase.hpp`? Maybe we should rename it.

Good point. Options that I have thought about are:

- `physicsPackageStep()`: pretty self explanatory that it is a "step" for the physics package.
- `timeStep()`: From the codes point of view it is a timestep even if there is no significance on `dt` such as in a quasi-static mechanics step, or equilibrium heat flow.
- `physicsPackageIncrement()`: detach from the "timestep" and just call it an increment. The event manager kind of imposes the timestep, but perhaps the physics package shouldn't care. And in relativistic physics, the passing of time is not constant on all clocks.


* I am going with `simulationStep()`

**Pull request closed without merging**